R/proActiv.R

```diff
@@ -0,0 +1,89 @@
+#' Wrapper function returning Summarized Experiment object giving promoter counts and activity
+#'
+#' @param promoterAnnotationData A PromoterAnnotation object containing the
+#'   reduced exon ranges, annotated intron ranges, promoter coordinates and the
+#'   promoter id mapping
+#' @param filePaths A character vector. The list of input files for 
+#'   which the junction read counts will be calculated
+#' @param fileLabels A character vector. The labels of input files 
+#'   for which the junction read counts will be calculated. These labels will be 
+#'   used as column names for each output data.frame object. If not provided,
+#'   filenames will be used as labels. Defaults to NULL
+#' @param genome A character. Genome version. Must be specified if input file
+#'   type is a BAM file. Defaults to NULL
+#' @param numberOfCores A numeric value. The number of cores to be used for 
+#'   counting junction reads. Defaults to 1 (no parallelization). This parameter 
+#'   will be used as an argument to BiocParallel::bplapply
+#' 
+#'
+#' @export
+#' @return A SummarizedExperiment object with assays giving promoter counts and activity 
+#'   with gene expression stored as column data and promoter gene id mapping stored as 
+#'   row data
+#' 
+#' @examples
+#' \dontrun{
+#' filePaths <- c('./sample1-tophat.bed', './sample2-tophat.bed')
+#' proActivFromJunctions <- proActiv(promoterAnnotationData = promoterAnnotationData, 
+#'                                  filePaths = filePaths, 
+#'                                  fileLabels = NULL, 
+#'                                  genome = NULL, 
+#'                                  numberOfCores = 1)
+#' 
+#' filePaths <- c('./sample1.bam', './sample2.bam')
+#' proActivFromBAM <- proActiv(promoterAnnotation Data = promoterAnnotationData,
+#'                            filePaths = filePaths,
+#'                            fileLabels = NULL,
+#'                            genome = 'hg19',
+#'                            numberOfCores = 1)}
+#'                            
+#' @import SummarizedExperiment
+#' @import S4Vectors
+#'
+proActiv <- function(promoterAnnotationData, filePaths = NULL, fileLabels = NULL, 
```

Comment:
(copied form original comment) maybe simplify argument names to:
function(files, promoterAnnotation, fileLabels = NULL, genome = NULL, ncores = 1)

Reply:
ok! should this change be propagated through the entire code base? (in particular, promoterAnnotationData to promoterAnnotation)

Reply:
the "Data" seems unnecessary, but it will break any code that has used the old version if we change this outside of proActiv() right? Maybe we just change this for proActiv() first. If we want to make changes to the other argument names we can make that separately?